frontend/lib/src/components/widgets/DataFrame/DataFrame.tsx

```diff
@@ -186,7 +186,10 @@ function DataFrame({
 
   // Number of rows of the table minus 1 for the header row:
   const dataDimensions = data.dimensions
-  const originalNumRows = Math.max(0, dataDimensions.rows - 1)
+  // data.columns refers to the header rows, and
+  // not the data columns. Not sure why it is named this way.
+  const numHeaderRows = data.columns.length
```

Comment:
nit: I would move the variable declaration right above [line 528](https://github.com/streamlit/streamlit/pull/9483/files#diff-eda1f0faafab9bbafe2c7b8b34241c35192488443ab3e02f0291d91e40ef632fR528) where it is used for the first time

Reply:
I removed the variable itself since its only used one time in the latest state anyways.